modules/xmpp/TraceablePeerConnection.js

```diff
@@ -593,26 +658,62 @@ TraceablePeerConnection.prototype.createAnswer
     this.trace('createAnswer', JSON.stringify(constraints, null, ' '));
     this.peerconnection.createAnswer(
         function (answer) {
-            self.trace('createAnswerOnSuccess::preTransform', dumpSDP(answer));
+
+            if (self.history) {
+                // Add a point in history.
+                var point = {
+                    action: 'createAnswer',
+                    transformations: []
+                };
+
+                // Add a transformation in the point.
+                point.transformations.push({
+                    name: 'createAnswer',
+                    sessionDescription: answer
+                });
+
+                self.history.push(point);
+            }
+
             // if we're running on FF, transform to Plan A first.
             if (RTCBrowserType.usesUnifiedPlan()) {
                 answer = self.interop.toPlanB(answer);
-                self.trace('createAnswerOnSuccess::postTransform (Plan B)',
-                    dumpSDP(answer));
+
+                if (self.history) {
+                    // Add a transformation in the point.
+                    var point = self.history[self.history.length - 1];
+                    point.transformations.push({
+                        name: 'toPlanB',
+                        sessionDescription: answer
+                    });
+                }
             }
 
-            if (!self.session.room.options.disableSimulcast
+            if (self.numOfSimulcastLayers > 1
                 && self.simulcast.isSupported()) {
                 answer = self.simulcast.mungeLocalDescription(answer);
-                self.trace('createAnswerOnSuccess::postTransform (simulcast)',
-                    dumpSDP(answer));
+
+                if (self.history) {
+                    // Add a transformation in the point.
+                    var point = self.history[self.history.length - 1];
+                    point.transformations.push({
+                        name: 'simulcast',
+                        sessionDescription: answer
+                    });
+                }
             }
 
             if (!RTCBrowserType.isFirefox())
             {
                 answer = self.ssrcReplacement(answer);
-                self.trace('createAnswerOnSuccess::mungeLocalVideoSSRC',
-                    dumpSDP(answer));
+
+                if (self.history) {
+                    // Add a transformation in the point.
+                    point.transformations.push({
```

Comment:
Yup, as @bgrozev said, the multiple var declarations are automatically hoisted by JS into one to the top so it doesn't matter (to JS) where the 'var point's  are and a single one anywhere in the function is enough.